app/src/main/java/io/mochahub/powermeter/workoutsession/workoutsessiondialog/NewWorkoutDialog.kt

```diff
@@ -0,0 +1,340 @@
+import android.os.Bundle
+import android.util.Log
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import androidx.fragment.app.DialogFragment
+import androidx.navigation.fragment.navArgs
+import androidx.recyclerview.widget.ItemTouchHelper
+import com.airbnb.epoxy.EpoxyTouchHelper
+import io.mochahub.powermeter.R
+import io.mochahub.powermeter.data.AppDatabase
+import io.mochahub.powermeter.data.ExerciseEntity
+import io.mochahub.powermeter.models.Exercise
+import io.mochahub.powermeter.models.Workout
+import io.mochahub.powermeter.models.WorkoutSession
+import io.mochahub.powermeter.models.WorkoutSet
+import io.mochahub.powermeter.models.addSet
+import io.mochahub.powermeter.models.removeSet
+import io.mochahub.powermeter.models.setReps
+import io.mochahub.powermeter.models.setWeight
+import io.mochahub.powermeter.models.updateExercise
+import kotlinx.android.synthetic.main.dialog_new_workout.addWorkoutBtn
+import kotlinx.android.synthetic.main.dialog_new_workout.newWorkoutDateText
+import kotlinx.android.synthetic.main.dialog_new_workout.newWorkoutNameText
+import kotlinx.android.synthetic.main.dialog_new_workout.newWorkoutToolbar
+import kotlinx.android.synthetic.main.fragment_exercise.recyclerView
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.launch
+import java.time.Instant
+import java.util.Calendar
+import java.util.Date
+import kotlin.math.absoluteValue
+
+class NewWorkoutDialog : WorkoutController.AdapterCallbacks, DialogFragment() {
+
+    private val args: NewWorkoutDialogArgs by navArgs()
+    private var shouldSave = true
+    private lateinit var workoutController: WorkoutController
+    private lateinit var viewModel: NewWorkoutViewModel
+    private var exercises = listOf<ExerciseEntity>()
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setStyle(
+            STYLE_NORMAL,
+            R.style.FullScreenDialog
+        )
+
+        viewModel =
+            NewWorkoutViewModel(
+                db = AppDatabase(requireContext())
+            )
+        CoroutineScope(Dispatchers.IO).launch {
+            exercises = viewModel.getExercises()
+        }.invokeOnCompletion {
+            workoutController =
+                WorkoutController(
+                    requireContext(),
+                    exercises.map { it.name },
+                    this
+                )
+        }
+    }
+
+    override fun onStart() {
+        super.onStart()
+
+        if (dialog != null) {
+            val width = ViewGroup.LayoutParams.MATCH_PARENT
+            val height = ViewGroup.LayoutParams.MATCH_PARENT
+            dialog!!.window!!.setLayout(width, height)
+        }
+    }
+
+    override fun onConfigurationChanged(newConfig: Configuration) {
+        super.onConfigurationChanged(newConfig)
+        workoutController.setData(viewModel.workouts)
+    }
```

Comment:
Not ideal

Reply:
I think we can just set an observer on `viewModel.workouts` that does `workoutController.setData()` right?

Reply:
Removed here https://github.com/power-meter/android-client/pull/32/commits/67e4c8c935c5f1e1c08d38f71487f2132c61b037 .  Still doesn't solve rotation, will have to look at how to fix it without using `configChange`